buffer/src/main/java/io/netty/buffer/api/CompositeBuffer.java

```diff
@@ -0,0 +1,2004 @@
+        if (closed) {
+            return bufferIsClosed(this);
+        }
+        if (write && readOnly) {
+            return bufferIsReadOnly(this);
+        }
+        return new IndexOutOfBoundsException(
+                "Index " + index + " is out of bounds: [read 0 to " + woff + ", write 0 to " +
+                capacity + "].");
+    }
+
+    private BufferAccessors chooseBuffer(int index, int size) {
+        int i = searchOffsets(index);
+        if (i == bufs.length) {
+            // This happens when the read/write offsets are parked 1 byte beyond the end of the buffer.
+            // In that case it should not matter what buffer is returned, because it shouldn't be used anyway.
+            return null;
+        }
+        int off = index - offsets[i];
+        Buffer candidate = bufs[i];
+        if (off + size <= candidate.capacity()) {
+            subOffset = off;
+            return candidate;
+        }
+        subOffset = index;
+        return tornBufAccessors;
+    }
+
+    private BufferAccessors choosePassThroughBuffer(int index) {
+        int i = searchOffsets(index);
+        return bufs[i];
+    }
+
+    private int searchOffsets(int index) {
+        int i = Arrays.binarySearch(offsets, index);
+        return i < 0? -(i + 2) : i;
+    }
+
+    // <editor-fold defaultstate="collapsed" desc="Torn buffer access.">
+    private static final class TornBufferAccessors implements BufferAccessors {
+        private final CompositeBuffer buf;
+
+        private TornBufferAccessors(CompositeBuffer buf) {
+            this.buf = buf;
+        }
+
+        @Override
+        public byte readByte() {
+            throw new AssertionError("Method should not be used.");
+        }
+
+        @Override
+        public byte getByte(int roff) {
+            throw new AssertionError("Method should not be used.");
+        }
+
+        @Override
+        public int readUnsignedByte() {
+            throw new AssertionError("Method should not be used.");
+        }
+
+        @Override
+        public int getUnsignedByte(int roff) {
+            throw new AssertionError("Method should not be used.");
+        }
+
+        @Override
+        public Buffer writeByte(byte value) {
+            throw new AssertionError("Method should not be used.");
+        }
+
+        @Override
+        public Buffer setByte(int woff, byte value) {
+            throw new AssertionError("Method should not be used.");
+        }
+
+        @Override
+        public Buffer writeUnsignedByte(int value) {
+            throw new AssertionError("Method should not be used.");
+        }
+
+        @Override
+        public Buffer setUnsignedByte(int woff, int value) {
+            throw new AssertionError("Method should not be used.");
+        }
+
+        @Override
+        public char readChar() {
+            if (bigEndian()) {
+                return (char) (read() << 8 | read());
+            } else {
```

Comment:
nit: when you return in an if block you can remove the `else`... this is true for everywhere

Reply:
These methods are very repetitive, so I think the consistent indentation is nicer here.

Reply:
With configurable endianness gone, this is no longer relevant.